src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/GenerateOrderCallbackHandler.java

```diff
@@ -0,0 +1,143 @@
+        Map<String, Object> data = new HashMap<>();
+        data.put("directionList", ImmutableList.of(
+            CCDOrderDirectionType.DOCUMENTS.name(),
+            CCDOrderDirectionType.EYEWITNESS.name()
+        ));
+        data.put("docUploadDeadline", deadline);
+        data.put("eyewitnessUploadDeadline", deadline);
+        data.put("preferredCourt", ccdCase.getPreferredCourt());
+        addCourtData(ccdCase, data);
+        return AboutToStartOrSubmitCallbackResponse
+            .builder()
+            .data(data)
+            .build();
+    }
+
+    private CallbackResponse generateOrder(CallbackParams callbackParams) {
+        CallbackRequest callbackRequest = callbackParams.getRequest();
+        CCDCase ccdCase = jsonMapper.fromMap(
+            callbackRequest.getCaseDetailsBefore().getData(), CCDCase.class);
+        CCDOrderGenerationData ccdOrderGenerationData = jsonMapper.fromMap(
+            callbackRequest.getCaseDetails().getData(), CCDOrderGenerationData.class);
+
+        String authorisation = callbackParams.getParams()
+            .get(CallbackParams.Params.BEARER_TOKEN).toString();
+        DocAssemblyRequest docAssemblyRequest = DocAssemblyRequest.builder()
+            .templateId(templateId)
+            .outputType(OutputType.DOC)
+            .formPayload(docAssemblyTemplateBodyMapper.from(
+                ccdCase, ccdOrderGenerationData, userService.getUserDetails(authorisation)))
+            .build();
+
+        DocAssemblyResponse docAssemblyResponse = docAssemblyClient.generateOrder(
+            authorisation,
+            authTokenGenerator.generate(),
+            docAssemblyRequest
+        );
+        return AboutToStartOrSubmitCallbackResponse
+            .builder()
+            .data(ImmutableMap.of(
+                "draftOrderDoc",
+                CCDDocument.builder()
+                    .documentUrl(docAssemblyResponse.getRenditionOutputLocation())
+                    .build()
+            ))
+            .build();
+    }
+
+    private void addCourtData(CCDCase ccdCase, Map<String, Object> data) {
+        String newRequestedCourt = null;
+        String preferredCourtObjectingParty = null;
+        String preferredCourtObjectingReason = null;
+        CCDApplicant applicant = ccdCase.getApplicants().get(0).getValue();
+        CCDRespondent respondent = ccdCase.getRespondents().get(0).getValue();
+
+        if (StringUtils.isNotBlank(applicant.getPreferredCourtReason())) {
+            newRequestedCourt = applicant.getPreferredCourtName();
+            preferredCourtObjectingParty = "CLAIMANT";
+            preferredCourtObjectingReason = applicant.getPreferredCourtReason();
+        } else if (StringUtils.isNotBlank(respondent.getPreferredCourtReason())) {
+            newRequestedCourt = respondent.getPreferredCourtName();
+            preferredCourtObjectingParty = "DEFENDANT";
+            preferredCourtObjectingReason = respondent.getPreferredCourtReason();
+        }
```

Comment:
The preferred court depends on if the defendant is a business; is something missing here?

Reply:
this is just picking what it is stored, I think it will be whoever saves the DQ stuff in the db to map things correctly